src/main/java/com/myhexin/cragent/coderabbittest/Counter.java

```diff
@@ -0,0 +1,29 @@
+package com.myhexin.cragent.coderabbittest;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Counter {
+
+    private AtomicInteger current = new AtomicInteger();
+
+    private final int sampleRate;
+
+    private final int checkValue;
+
+    public Counter(int sampleRate, int checkValue) {
+        this.sampleRate = sampleRate;
+        this.checkValue = checkValue;
+    }
+
+    public boolean checkAndInc() {
+        // 原子获取v
+        int v = current.getAndAccumulate(1, (pre, up) -> {
+            int next = pre + up;
+            if (next >= sampleRate) {
```

Comment:
**Issue**: The method 'getAndAccumulate' resets the counter to 0 if it reaches the 'sampleRate', which might not be thread-safe if multiple threads reach this condition around the same time. <br> **Fix**: Consider using a loop with compareAndSet to ensure atomicity during the reset operation. <br> **Code Suggestion**: 
 ```
 -            if (next >= sampleRate) {
 -                return 0;
 +            if (next >= sampleRate) {
 +                if (current.compareAndSet(next, 0)) {
 +                    return 0;
 +                } else {
 +                    return next;
 +                }
 ```